addons/binding/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/handler/PowermaxBridgeHandler.java

```diff
@@ -0,0 +1,752 @@
+                    PowermaxCommDriver.getTheCommDriver().sendMessage(PowermaxSendType.RESTORE);
+                    currentState.setLastKeepAlive(now);
+                }
+
+                //
+                // Try to reconnect if disconnected
+                //
+
+                if (!connected) {
+                    logger.debug("trying to reconnect...");
+                    closeConnection();
+                    currentState = new PowermaxState();
+                    openConnection();
+                    if (connected) {
+                        updateStatus(ThingStatus.ONLINE);
+                        if (forceStandardMode) {
+                            currentState.setPowerlinkMode(false);
+                            updateChannelsFromAlarmState(MODE, currentState);
+                            settings = PowermaxPanelSettings.getThePanelSettings();
+                            if (settings.process(false, panelType, null)) {
+                                for (PowermaxPanelSettingsListener listener : listeners) {
+                                    listener.onPanelSettingsUpdated(settings);
+                                }
+                            }
+                            remainingDownloadTry = 0;
+                            updatePropertiesFromPanelSettings();
+                            logger.info("Powermax alarm binding: running in Standard mode");
+                            PowermaxCommDriver.getTheCommDriver().sendMessage(PowermaxSendType.ZONESNAME);
+                            PowermaxCommDriver.getTheCommDriver().sendMessage(PowermaxSendType.ZONESTYPE);
+                            PowermaxCommDriver.getTheCommDriver().sendMessage(PowermaxSendType.STATUS);
+                        } else {
+                            PowermaxCommDriver.getTheCommDriver().startDownload();
+                        }
+                    } else {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "reconnection failed");
+                    }
+                } else if ((remainingDownloadTry > 0) && !PowermaxCommDriver.getTheCommDriver().isDownloadRunning()
+                        && ((PowermaxCommDriver.getTheCommDriver().getLastTimeDownloadRequested() == null) || ((now
+                                - PowermaxCommDriver.getTheCommDriver().getLastTimeDownloadRequested()) >= 45000))) {
+                    // We wait at least 45 seconds before each retry to download the panel setup
+                    logger.info("Powermax alarm binding: try again downloading setup");
+                    PowermaxCommDriver.getTheCommDriver().startDownload();
+                }
+
+            } catch (Exception e) {
+                logger.warn("Exception in scheduled job: {}", e.getMessage(), e);
+            } catch (Error e) {
+                logger.warn("Error in scheduled job: {}", e.getMessage(), e);
+            } catch (Throwable t) {
```

Comment:
Just a security to be sure that the refresh job will never be stopped to be rescheduled.
I know that it is not the clean way but this is not clear how to implement in such a way that I will be sure that my job will always continue to be scheduled.

Reply:
If `Error`s or `Throwable`s are thrown from the body of the `Runnable` it should terminate note that we log that errors nowadays because we wrap all the tasks we schedule to make sure our tasks do not stop without mentioning the cause.

So please remove the `Error` / `Throwable` catching.

Reply:
Done but you will hear about me if I discover in the future that the job sometimes stopped to be scheduled ;)